linux/adv/packet.go

```diff
@@ -219,8 +219,28 @@ func (p *Packet) ServiceSol() []ble.UUID {
 
 // ServiceData ...
 func (p *Packet) ServiceData() []ble.ServiceData {
-	v, _ := p.m[keys.serviceData].([]ble.ServiceData)
-	return v
+	m, ok := p.m[keys.serviceData].(map[string][]interface{})
+	if !ok {
+		return nil
+	}
+
+	// map -> array
+	arr := []ble.ServiceData{}
+	for su, arr := range m {
```

Comment:
Why is `arr` being using as both the function return an an assigned loop variable?